presto-main/src/test/java/com/facebook/presto/operator/TestAnnotationEngineForSqlInvokedScalars.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.common.QualifiedObjectName;
+import com.facebook.presto.common.type.ArrayType;
+import com.facebook.presto.common.type.TypeSignature;
+import com.facebook.presto.metadata.FunctionAndTypeManager;
+import com.facebook.presto.operator.scalar.annotations.SqlInvokedScalarFromAnnotationsParser;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.FunctionKind;
+import com.facebook.presto.spi.function.Signature;
+import com.facebook.presto.spi.function.SqlInvokedFunction;
+import com.facebook.presto.spi.function.SqlInvokedScalarFunction;
+import com.facebook.presto.spi.function.SqlParameter;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+import com.facebook.presto.spi.function.TypeVariableConstraint;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.metadata.BuiltInTypeAndFunctionNamespaceManager.DEFAULT_NAMESPACE;
+import static com.facebook.presto.metadata.FunctionAndTypeManager.createTestFunctionAndTypeManager;
+import static com.facebook.presto.spi.function.SqlFunctionVisibility.PUBLIC;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestAnnotationEngineForSqlInvokedScalars
+        extends TestAnnotationEngine
+{
+    private static final FunctionAndTypeManager FUNCTION_AND_TYPE_MANAGER = createTestFunctionAndTypeManager();
+
+    public static class SingleImplementationSQLInvokedScalarFunction
```

Comment:
- private
- move the helper function to the end of the file

Reply:
> * private

can't make them private because it needs to be accessed through reflection by an external class (similar classes in the other test file are also declared public e.g. [here](https://github.com/prestodb/presto/blob/f85a1c2bf1bcda217655c24220059bc5377d6b76/presto-main/src/test/java/com/facebook/presto/operator/TestAnnotationEngineForScalars.java#L216-L225))

```
com.facebook.presto.spi.PrestoException: Failed to get function body for method [public static java.lang.String com.facebook.presto.operator.TestAnnotationEngineForSqlInvokedScalars$SingleImplementationSQLInvokedScalarFunction.fun()]
	at com.facebook.presto.operator.scalar.annotations.SqlInvokedScalarFromAnnotationsParser.createSqlInvokedFunctions(SqlInvokedScalarFromAnnotationsParser.java:159)
Caused by: java.lang.IllegalAccessException: Class com.facebook.presto.operator.scalar.annotations.SqlInvokedScalarFromAnnotationsParser can not access a member of class com.facebook.presto.operator.TestAnnotationEngineForSqlInvokedScalars$SingleImplementationSQLInvokedScalarFunction with modifiers "public static"
	at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:102)
	at java.lang.reflect.AccessibleObject.slowCheckMemberAccess(AccessibleObject.java:296)
	at java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:288)
	at java.lang.reflect.Method.invoke(Method.java:491)
```

> * move the helper function to the end of the file

done